File: eform-client/src/app/plugins/modules/rentable-items-pn/models/contract.model.ts
import {RentableItemPnModel} from './rentable-item-pn.model';
import {RentableItemCustomerModel} from 'src/app/plugins/modules/rentable-items-pn/models/rentableItemCustomerModel';

export class ContractModel {
  id: number;
  contractStart: Date;
  contractEnd: Date;
  rentableItemCustomer: RentableItemCustomerModel;
  customerId: number;
  contractNr: number;
  rentableItems: Array<RentableItemPnModel> = [];
  rentableItemIds: Array<number> = [];
  deleteIds: Array<number> = [];
}

File: eFormAPI/Plugins/RentableItems.Pn/RentableItems.Pn/Infrastructure/Models/ContractModel.cs
﻿using System;
using System.Collections.Generic;
using RentableItems.Pn.Infrastructure.Models.Customer;

namespace RentableItems.Pn.Infrastructure.Models
{
    public class ContractModel
    {
        public int Id { get; set; }
        public string WorkflowState { get; set; }
        public int? Version { get; set; }
        public DateTime? CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
        public int CreatedByUserID { get; set; }
        public int UpdatedByUserID { get; set; }
        public DateTime? ContractStart { get; set; }
        public DateTime? ContractEnd { get; set; }
        public int CustomerId { get; set; }
        public RentableItemCustomerModel RentableItemCustomer { get; set; }
        public int? ContractNr { get; set; }
        public List<int> RentableItemIds { get; set; }
        public List<RentableItemModel> RentableItems { get; set; }
        public List<int> DeleteIds { get; set; }
    }
}
